catchup/service.go

```diff
@@ -201,6 +236,10 @@ func (s *Service) innerFetch(r basics.Round, peer network.Peer) (blk *bookkeepin
 //  - If the block is already in the ledger (e.g. if agreement service has already written it)
 //  - If the retrieval of the previous block was unsuccessful
 func (s *Service) fetchAndWrite(r basics.Round, prevFetchCompleteChan chan bool, lookbackComplete chan bool, peerSelector *peerSelector) bool {
+	// If sync-ing this round would break our cache invariant, don't fetch it
+	if syncRound, err := s.GetSyncRound(); err == nil && syncRound != 0 && r >= basics.Round(syncRound+s.cfg.MaxAcctLookback) {
```

Comment:
Interesting thought. I don't see a reason why we couldn't allow this, but designing this around Indexer's use case it wouldn't make sense for Indexer to ever go backwards in rounds.

I think you're right though that we should make it as permissive as possible--I'll update.

Reply:
On second thought, I'm not sure this is a good idea...

The catchup service has no way of knowing what is in the cache which means we may or may not have the last `MaxAcctLookback` rounds in the cache. For example, when you first start the node you will have no deltas cached. If you were to then set the sync round to Latest - 2 and the API succeeded, you would assume that you can fetch Latest-1, but it would not be there.

I think erring on the side of caution here is best, which would mean reducing the set of circumstances in which users request a delta and it's not available.

Reply:
That's fair.

I'm now wondering about fetching deltas. I haven't checked the code yet, but I assume that right now, we allow fetching deltas for any round that still in cache. I wonder if we should explicitly and only allow it for the syncround.  That way there's no "luck". You don't get to ask for an older round and have it work at first, and then stop working.

Reply:
Yes, currently it allows for anything in the cache--I'll think about whether or not to revise that to "just the sync round" in the set of revisions I'm working on currently. 

Not sure if this is useful since we don't currently batch block retrievals during Indexer catchup, but a thought I had was to adjust cache size allowing for future batching APIs. That wouldn't be possible if we restrict to `fetch round == sync round`.